eagle-jpm/eagle-jpm-util/src/main/java/org/apache/eagle/jpm/util/Utils.java

```diff
@@ -116,6 +116,6 @@ public static long parseMemory(String memory) {
 
     public static String makeLockPath(String zkrootWithSiteId) {
         Preconditions.checkArgument(StringUtils.isNotBlank(zkrootWithSiteId), "zkrootWithSiteId must not be blank");
-        return zkrootWithSiteId.toLowerCase() + "/locks";
+        return "/locks" + zkrootWithSiteId.toLowerCase();
```

Comment:
As I mentioned, if we have structure like 
zkRoot/locks
zkRoot/jobs/1
zkRoot/jobs/2
yarnAppIds will be curator.getChildren().forPath(this.zkRoot + "/jobs");
so, you need to do a simple refactor